### ISSUE TITLE
fetch_data to submit a batch job via URL not working 

### ISSUE THREAD

**Issue by nickgerig** [speechmatics/speechmatics-js-sdk#49]
**Describe the bug**
Calling `fetch_data` as per the [specification](https://docs.speechmatics.com/features-other/fetch-url#configuration-example) does not work in the JS SDK.

**To Reproduce**

```sm.batch
  .transcribe({
    fetch_data: { url: '<url>` },
    transcription_config: { language: 'en' },
  })
  .then(({ results }) => {
    console.log(results.map((r) => r.alternatives[0].content).join(' '));
  })
  .catch((error) => {
    console.log(error);
    process.exit(1);
  });
```

 - Version [3.1.7]

**Comment by nickgerig** [comment]
This will be fixed shortly with an upcoming major release. In the meantime please use this workaround:

```
sm.batch.transcribe({
  input: {
    fetch: {
      url: '<url>',
    },
  },
});
```

**Comment by erkkimon** [comment]
Thank you. Any guess when the next release is coming up? Are we talking about days or weeks?

**Comment by nickgerig** [comment]
We plan to do it next week

**Comment by nickgerig** [comment]
I'll close this bug once it's released so you should get notified.

**Comment by erkkimon** [comment]
I appreciate it, cheers @nickgerig!

**Comment by mnemitz** [comment]
Hi @erkkimon, we've just released version 4.0.0 of the SDK in which we've changed the function signature for `transcribe(...)` to more clearly distinguish between the URL and File cases. See the release notes here: https://github.com/speechmatics/speechmatics-js-sdk/releases/tag/v4.0.0

We've also updated the examples and added a new one for URL fetching: https://github.com/speechmatics/speechmatics-js-sdk/blob/main/examples/example_batch_url.js

**Comment by willemmulder** [comment]
@nickgerig I don't want to hijack this issue, but I was just wondering if there is any thoughts on supporting realtime streams using URLs as well? We use Speechmatics to transcribe m3u of mp3 livestream, and it would be great if the stream-reading part could live on the Speechmatics side. That would prevent us from having to fetch the stream, chunk it and forward it to Speechmatics, with all extra processing and potential errors that come with that.
